src/Components/Recommendation/Recommendation.cy.js

```diff
@@ -109,6 +118,101 @@ describe('recommendation page for enabled recommendation with clusters enabled a
+    cy.ouiaId('hosts-acked')
+      .ouiaId('enable')
+      .click()
+      .then(() => {
+        // wait for the expected number of network calls
+        for (const cluster of clusterDetails.data.disabled) {
+          cy.wait('@enableRequest').then((xhr) => {
+            expect(xhr.request.url).to.contain(cluster.cluster_id);
+          });
+        }
+      });
+  });
+
+  describe('disabled clusters modal', () => {
+    it('can be displayed', () => {
+      cy.ouiaId('hosts-acked').ouiaId('view-clusters').click();
+      cy.ouiaId('hosts-disabled').should('exist');
+    });
+    it('has expected clusters', () => {
+      cy.ouiaId('hosts-acked').ouiaId('view-clusters').click();
+      cy.ouiaId('hosts-disabled')
+        .find(TBODY)
+        .find(ROW)
+        .should('have.length', clusterDetails.data.disabled.length);
+      const clusters = _.map(
+        clusterDetails.data.disabled,
+        (it) => it.cluster_name || it.cluster_id
+      );
+      cy.get(`td[data-label="Cluster name"]`)
+        .then(($els) => {
+          return _.map(Cypress.$.makeArray($els), 'innerText');
+        })
+        .should('deep.equal', clusters);
+    });
+  });
+});
+
+describe('recommendation page for enabled recommendation without disabled clusters', () => {
+  let data = _.cloneDeep(clusterDetails.data);
+  data.disabled = [];
+  beforeEach(() => {
+    mount(
+      <MemoryRouter>
+        <Intl>
+          <Provider store={getStore()}>
+            <Recommendation
+              rule={{ ...defaultPropsRule }}
+              ack={{ ...defaultPropsAck, data: undefined }}
+              clusters={{ ...defaultPropsClusterDetails, data: data }}
+              match={{ params: { recommendationId: 'X' } }}
+            />
+          </Provider>
+        </Intl>
+      </MemoryRouter>
+    );
+
+    cy.intercept('POST', '/api/insights-results-aggregator/v2/ack', {
+      statusCode: 201,
+    }).as('ackRequest');
+  });
+
+  it('header shows description', () => {
+    cy.ouiaType('PF4/Title', 'h1')
+      .should(($el) => expect($el.text().trim()).to.equal(ruleDescription))
+      .and('have.length', 1);
+  });
+
+  it('does not show info about disabled clusters', () => {
+    cy.ouiaId('hosts-acked').should('not.exist');
+  });
+
+  it('table is displayed', () => {
+    cy.get('#affected-list-table')
+      .within(() => {
+        cy.ouiaId('clusters').should('have.length', 1);
+      })
+      .parent()
+      .ouiaType('PF4/Title', 'h3')
+      .should('have.text', 'Affected clusters');
+  });
+
+  it('category labels are displayed', () => {
+    cy.get('.categoryLabels').should('have.length', 1);
+  });
+
+  it('actions can ack recommendation', () => {
+    cy.ouiaId('actions')
+      .click()
+      .within(() => {
+        cy.get('a').should('have.text', 'Disable recommendation').click();
+      });
+    cy.ouiaId('recommendation-disable').should('exist');
+  });
 });
 
 // TODO check disabledRule with ack data undefined. Makes sense?
```

Comment:
@gkarat @Fewwy Do you think it makes sense to test this scenario, or we can skip it?